### PR TITLE
position-delete optimize

### PR DESCRIPTION
**Pull request by shidayang**
reuse position delete for node

### CODE REVIEW

File: trino/src/test/java/com/netease/arctic/trino/iceberg/ArcticQueryRunnerForClient.java [hunk 1]
```diff
@@ -61,7 +61,8 @@ public static DistributedQueryRunner createIcebergQueryRunner(
 
         queryRunner.installPlugin(new ArcticPlugin());
         Map<String, String> icebergProperties = ImmutableMap.<String, String>builder()
-            .put("arctic.url", "thrift://localhost:8080/" + ARCTIC_CATALOG)
+            // .put("arctic.url", "thrift://localhost:8080/" + ARCTIC_CATALOG)
+            .put("arctic.url", "thrift://10.196.98.23:18111/trino_online_env")
```
Comment (hameizi) [inline trino/src/test/java/com/netease/arctic/trino/iceberg/ArcticQueryRunnerForClient.java:65]: private ip

Comment (shidayang) [reply trino/src/test/java/com/netease/arctic/trino/iceberg/ArcticQueryRunnerForClient.java:65]: 。。。。

File: core/src/main/java/com/netease/arctic/io/reader/ArcticDeleteFilter.java [hunk 2]
```diff
@@ -58,11 +64,22 @@
 
 /**
  * Abstract implementation of filtering equality delete files with insert files and base files.
```
Comment (zhoujinsong) [inline core/src/main/java/com/netease/arctic/io/reader/ArcticDeleteFilter.java:66]: It seems the class description should be updated

Comment (shidayang) [reply core/src/main/java/com/netease/arctic/io/reader/ArcticDeleteFilter.java:66]: ok

File: core/src/main/java/com/netease/arctic/io/reader/BaseIcebergDataReader.java [hunk 3]
```diff
@@ -107,6 +107,19 @@ public CloseableIterable<T> readData(ArcticFileScanTask task) {
     return iterable;
   }
 
+  public CloseableIterable<T> readDataWithoutApplyPosDelete(ArcticFileScanTask task) {
```
Comment (zhoujinsong) [inline core/src/main/java/com/netease/arctic/io/reader/BaseIcebergDataReader.java:110]: the new method makes this class very strange. functions use the old `readData` method still have bad performace like flink reader?

Comment (shidayang) [reply core/src/main/java/com/netease/arctic/io/reader/BaseIcebergDataReader.java:110]: The readData method is used by original iceberg read

**Review by zhoujinsong — APPROVED** [review]

**Pull request merged**